Cosmostation/Controller/Setting/SettingTableViewController.swift

```diff
@@ -514,6 +525,56 @@ class SettingTableViewController: UITableViewController, PasswordViewDelegate, Q
         }
     }
     
+    func onShowLanguageDialog() {
+        let showAlert = UIAlertController(title: nil, message: nil, preferredStyle: .alert)
+        showAlert.overrideUserInterfaceStyle = BaseData.instance.getThemeType()
+        let systemAction = UIAlertAction(title: "System", style: .default, handler: { _ in
+            self.onSetLanguage(0)
```

Comment:
this hardcoded numbers setting the language can be problematic, I think that adding semantics to these numbers will increase readability and can avoid maintenance issues in the future
```swift
enum Language: Int, CustomStringConvertible {
  case system 
  case english
  case japanese
  case korean

  var description: String {
     switch self {
         case system: return Locale.preferredLanguages.first ?? """
         case english: return "en""
 
         ....
     }
  }
}
```

Reply:
it will apply also to `BaseData`
`func setLanguage(lang: Language)`and `func getLanguage() -> Language`